include/mbedtls/build_info.h

```diff
@@ -74,6 +74,9 @@
 #include MBEDTLS_USER_CONFIG_FILE
 #endif
 
+#if defined(MBEDTLS_PK_C) && defined(MBEDTLS_USE_PSA_CRYPTO)
+#define MBEDTLS_PK_WRITE_C
```

Comment:
> I'm not entirely convinced that making a precedent by turning on a config option silently in build_info.h is good.

It's something we wanted to be able to do when we split the config file in two in the 3.0 release. The goal was to allow dependencies between options to change without breaking code, which seems to be the goal here. And we were already doing it in a specific scenario: to deduce MBEDTLS options from PSA options or vice versa in `mbedtls/config_psa.h`.

I'm not convinced that this specific case is correct, however. It seems too broad to me. In any case, please add a comment explaining why it's needed. And maybe it's only needed under some additional condition like RSA support or TLS support?

Reply:
Currently this is only needed when ECDSA is defined (see the version of `ecdsa_verify_wrap` under `MBEDTLS_USE_PSA_CRYTPO` but I expect to be also needed for RSA soon, which adds up to all of PK.

I think this dependency is suboptimal and I hope to perhaps lift it in the future: currently we're storing the key as an `ecdsa_keypair` (or `rsa_context`) even under `MBEDTLS_USE_PSA_CRYPTO`, so when we want to create a copy in a PSA key slot in order to do an operation with PSA, we need to serialize before calling `psa_key_import()`. It seems quite convoluted, so in the future I'd like to find something simpler, but there are a number of design questions to consider, so in the meantime I think this enables use to move forward.

Back to "is `build_info.h` the right place?": if we want to auto-enable things, sure, but we could also not auto-enable it and add a check in `check_config.h` instead. I thought auto-enable was more user friendly (especially for something that's about our current implementation strategy which may change in the future), but `check_config.h` works too. (There's no compatibility issue as it wouldn't introduce a new limitation, just document an existing one and turn a link-time error into a compile-time error.)